steps/redhat.py

```diff
@@ -72,3 +72,95 @@ def step_impl(context, host):
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep "Enabled the following repos" | cut -d ":" -f5 | sed "s/^ //"')[0]['stdout']
+	assert repoids_enabled == 'rhel-7-server-optional-beta-rpms, rhel-7-server-beta-debug-rpms', "Configured repoids weren't enabled"
+
+@then(u'check if it failed')
+def step_impl(context):
+	cloudinit_result = context.remote_cmd(cmd='shell',
+					module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | tail -n1 | cut -d ":" -f4 | sed "s/^ //"')[0]['stdout']
+	assert cloudinit_result == 'rh_subscription plugin did not complete successfully', 'rh_subscription plugin should have failed'
+
+@then(u'check if the subscription-manager failed to register with bad username')
+def step_impl(context):
+	register_result =  context.remote_cmd(cmd='shell',
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Invalid | cut -d ":" -f4 | sed -e "s/^ //" | tail -n1')[0]['stdout']
+	assert register_result == 'Invalid username or password. To create a login, please visit https', "subscription-manager didn't fail to register"
+
+@then(u'check if the subscription-manager failed to register with bad password')
+def step_impl(context):
+	register_result =  context.remote_cmd(cmd='shell',
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Invalid | cut -d ":" -f4 | sed -e "s/^ //" | tail -n1')[0]['stdout']
+	assert register_result == 'Invalid username or password. To create a login, please visit https', "subscription-manager didn't fail to register"
+
+@then(u'check if the subscription-manager failed to attach non-existent pool-id')
+def step_impl(context):
+	register_result =  context.remote_cmd(cmd='shell',
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Pool | cut -d ":" -f4 | sed -e "s/^ //"')[0]['stdout']
+	assert register_result == 'Pool 8a85f9823e3d5e43013e3ddd4e95ffff is not available', "Pool 8a85f9823e3d5e43013e3ddd4e95ffff shouldn't be available"
+
+@then(u'check if the subscription-manager failed to attach pool-id defined as a scalar')
+def step_impl(context):
+	register_result =  context.remote_cmd(cmd='shell',
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Pool | cut -d ":" -f4 | sed -e "s/^ //"')[0]['stdout']
+	assert register_result == 'Pools must in the format of a list.', "Pools in scalar form shouldn't be accepted"
+
+@then(u'check if an error message is shown in the log when trying to add non-existent repo')
+def step_impl(context):
+	register_result =  context.remote_cmd(cmd='shell',
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Repo | grep exist | cut -d ":" -f4 | sed -e "s/^ //"')[0]['stdout']
+	assert register_result == 'Repo rhel-7-server-beta-debug-rpm does not appear to exist', "Error message not found"
+
+@then(u'check the already enabled message')
```

Comment:
I would make this more specific...along the lines of "check for the repo already enabled message"